src/java/org/apache/cassandra/io/sstable/format/big/BigTableScanner.java

```diff
@@ -209,6 +209,48 @@ private void seekToCurrentRangeStart()
         }
     }
 
+    /**
+     * Gets the position in the data file, but does not seek to it.  This does seek the index to find the data position
+     * but does not actually seek the data file.
+     * @param position position to find in data file.
+     * @return
+     * @throws CorruptSSTableException
+     */
+    public long getDataPosition(PartitionPosition position) throws CorruptSSTableException {
```

Comment:
just to have less copy-pasted type thing can you change seekToCurrentRangeStart to use this?

Reply:
ie

```java
 // Helper method to seek to the index position and process entries
    private long seekAndProcess(PartitionPosition position, Range<PartitionPosition> range, boolean updateDataFile) throws CorruptSSTableException {
        long indexPosition = sstable.getIndexScanPosition(position);
        ifile.seek(indexPosition);

        try {
            while (!ifile.isEOF()) {
                indexPosition = ifile.getFilePointer();
                DecoratedKey indexDecoratedKey = sstable.decorateKey(ByteBufferUtil.readWithShortLength(ifile));

                boolean isFound = (range == null && indexDecoratedKey.compareTo(position) > 0)
                        || (range != null && (indexDecoratedKey.compareTo(range.left) > 0 || range.contains(indexDecoratedKey)));

                if (isFound) {
                    long dataPosition = RowIndexEntry.Serializer.readPosition(ifile);
                    ifile.seek(indexPosition);
                    if (updateDataFile) {
                        dfile.seek(dataPosition);
                    }
                    return dataPosition;
                } else {
                    RowIndexEntry.Serializer.skip(ifile, sstable.descriptor.version);
                }
            }
        } catch (IOException e) {
            sstable.markSuspect();
            throw new CorruptSSTableException(e, sstable.getFilename());
        }

        // If no position is found, return 0
        return 0L;
    }

    /**
     * Gets the position in the data file, but does not seek to it. This does seek the index to find the data position
     * but does not actually seek the data file.
     * @param position position to find in data file.
     * @return
     * @throws CorruptSSTableException
     */
    public long getDataPosition(PartitionPosition position) throws CorruptSSTableException {
        return seekAndProcess(position, null, false);
    }

    /**
     * Seeks to the start of the current range and updates both the index and data file positions.
     */
    private void seekToCurrentRangeStart() throws CorruptSSTableException {
        seekAndProcess(currentRange.left, currentRange, true);
    }
```

Reply:
makes sense, refactored as suggested 👍